### ISSUE TITLE
Best practice for removing entites based on some system logic?

### ISSUE THREAD

**Issue by joeblubaugh** [lovetoys/lovetoys#76]
I have a system that processes some entities, and then I would like to remove these entities once they have been processed.

Should I pass the engine as a parameter to my system so it can call `engine.removeEntity`, or should I use some other approach to removal? I think it's interesting that entities can deactivate themselves, but cannot remove themselves from the engine.

**Comment by Nukesor** [comment]
I think that's a good point.
We always developed games with the engine as a global variable for easy access.

Maybe we could introduce a flag on the entity, which marks it as `toBeDeleted` on the next tick by the entity manager. Hm, But this could lead to annoying side effects on the next tick.

The thing is that we followed the paradigm that entities should be as stupid as possible. A simple class which is only designed to hold data (`Components`) and should implement as few logic as possible.

I'll talk with @raffomania, since the entity then needs a reference to the engine, which is something we is something we tried to strictly avoid. But it's a point that it's really inconvenient, if the engine is not a global variable.

**Comment by code0wl** [comment]
Hi everyone, any progress on this?

**Comment by Nukesor** [comment]
To be honest, development has a stalled quite a bit.
Both of us moved to other languages/technologies and we haven't properly worked on this library for a really long time.
I'm not sure if this feature will be added by us.
However, we would be happy to review and accept PR's or maybe even add new people to the organization!

**Comment by code0wl** [comment]
@Nukesor Thanks for the reply. I think I am happy with my solution using the features already implemented in your lib. 

I share your sentiment greatly. I really like Lua, but it's on a steady decline for a long time now. Once this game is done and is distributed on steam. I will most likely move to either Rust or C# for game development.

**Comment by raffomania** [comment]
If you're looking for more full-fledged alternatives to love, I can greatly recommend godot :) It also supports C# if that's your thing

**Comment by code0wl** [comment]
@raffomania Yeah I've had a brief look at it and followed this course on it on Udemy. https://www.udemy.com/course/godot/ 

Sadly the C# part was not mature enough to be finished in the course, so they left it opened on "todo".

This made me decide to switch to Monogame at the end. 

Thanks for the suggestion. From what I have learned and used in Godot. Its really awesome!
